### PR TITLE
Update test and perf env to use CairoMakie for output

### LINKED ISSUE #2431: Switch to `CairoMakie` and remove `Plots.jl`
Our environments bring in all sorts of plotting libraries. In turn, plotting libraries bring in all sorts of `jll`s. This makes our environments very complex and vulnerable to dependency problems and incompatibilities. 

In one of my branches, my version of HDF5 is pinned to 1.12 because GR is pinned to 0.53 because libtiff is pinned to 4.4 because the latest Plots.jl is not compatible with GR 0.73. 

`Plots.jl` is not used much (primarily for contour plots in the gravity wave tests), so I suggest we remove it for `CairoMakie`.

### PR DESCRIPTION
**Pull request by akshaysridhar**
Gravity wave tests, and perf scaling checks use Plots.jl - this has caused issues with some FFMPEG_jll dependencies recently. This PR updates all applicable tests to use CairoMakie with a simple set of helper functions and removes `Plots` from deps. 

Closes #2431

### CODE REVIEW

File: test/gravity_wave/non_orographic_gravity_wave/nogw_test_mima.jl [hunk 1]
```diff
@@ -203,72 +204,97 @@ vforcing_zonalave = dropdims(mean(vforcing, dims = 1), dims = 1)
...
+            gwfv_zonalave[:, :, it] .* 100,
+        levels = range(-100, 100; length = 20),
+        title,
+        p_loc = (2, 3),
     )
-    p = plot(c1, c2, c3, c4, c5, c6, layout = (2, 3), size = (1500, 800))
-    png(p, joinpath(output_dir, string(it) * ".png"))
+
+    # Save to disk
+    CairoMakie.save(joinpath(output_dir, string(it) * ".png"), fig)
```
Comment (Sbozzolo) [inline test/gravity_wave/non_orographic_gravity_wave/nogw_test_mima.jl:299]:
```suggestion
    CairoMakie.save(joinpath(output_dir, "$it.png"), fig)
```

File: test/gravity_wave/orographic_gravity_wave/ogwd_3d.jl [hunk 2]
```diff
@@ -406,43 +403,57 @@ end
...
+        Y = lat,
+        Z = gfdl_udt_topo[:, :, k, 1] .* 86400,
+        levels = range(-10, 10; length = 20),
+        title,
+        p_loc = (2, 1),
+        yreversed = false,
     )
-    uplot = plot(p1, p2, layout = (2, 1))
-    png(uplot, joinpath(output_dir, "uforcing_" * string(k) * ".png"))
+    CairoMakie.save(joinpath(output_dir, "uforcing_" * string(k) * ".png"), fig)
```
Comment (Sbozzolo) [inline test/gravity_wave/orographic_gravity_wave/ogwd_3d.jl:430]:
```suggestion
    CairoMakie.save(joinpath(output_dir, "uforcing_$k.png"), fig)
```

File: test/gravity_wave/orographic_gravity_wave/ogwd_3d.jl [hunk 3]
```diff
@@ -406,43 +403,57 @@ end
...
+        Y = lat,
+        Z = gfdl_vdt_topo[:, :, k, 1] .* 86400,
+        levels = range(-10, 10; length = 20),
+        title,
+        p_loc = (2, 1),
+        yreversed = false,
     )
-    vplot = plot(p1, p2, layout = (2, 1))
-    png(vplot, joinpath(output_dir, "vforcing_" * string(k) * ".png"))
+    CairoMakie.save(joinpath(output_dir, "vforcing_" * string(k) * ".png"), fig)
```
Comment (Sbozzolo) [inline test/gravity_wave/orographic_gravity_wave/ogwd_3d.jl:458]:
```suggestion
    CairoMakie.save(joinpath(output_dir, "vforcing_$k.png"), fig)
```

File: test/gravity_wave/orographic_gravity_wave/ogwd_3d.jl [hunk 4]
```diff
@@ -10,12 +10,9 @@ using ClimaCoreTempestRemap
 
 using Interpolations
 
-using Plots
-using ClimaCorePlots
-
 const FT = Float64
-
-include("../../post_processing/remap/remap_helpers.jl")
+include("../../../post_processing/remap/remap_helpers.jl")
```
Comment (Sbozzolo) [inline test/gravity_wave/orographic_gravity_wave/ogwd_3d.jl:14]: Why do we need the remapping here?

**Review by Sbozzolo — COMMENTED** [review]
I have trivial comments. If you are happy with the plots, I am happy too.

File: test/gravity_wave/orographic_gravity_wave/ogwd_3d.jl (hunk 4 above)
Comment (Sbozzolo) [reply test/gravity_wave/orographic_gravity_wave/ogwd_3d.jl:14]: Was looking at the wrong file. I get it.

**Review by Sbozzolo — APPROVED** [review]

**Comment by akshaysridhar** [comment]
Thanks @Sbozzolo ; waiting for the new-central PR to go through, then I'll rebase

**Comment by akshaysridhar** [comment]
> I have trivial comments. If you are happy with the plots, I am happy too.

I think the plots themselves could be more descriptive at a glance in the CI outputs; I want to remove the Plots deps and generate as close to the original images with the new setup as the first step, we can go through and clean these up following this PR. (The gravity wave outputs could also be standardized with the ClimaAnalysis in the future)

**Comment by akshaysridhar** [comment]
@sriharshakandala Can you please verify the changes in the perf env?

**Pull request merged**